examples/gno.land/p/demo/int256/arithmetic.gno

```diff
@@ -0,0 +1,187 @@
+package int256
+
+import (
+	"gno.land/p/demo/uint256"
+)
+
+func (z *Int) Add(x, y *Int) *Int {
+	z.initiateAbs()
+
+	neg := x.neg
+
+	if x.neg == y.neg {
+		// x + y == x + y
+		// (-x) + (-y) == -(x + y)
+		z.abs = z.abs.Add(x.abs, y.abs)
+	} else {
+		// x + (-y) == x - y == -(y - x)
+		// (-x) + y == y - x == -(x - y)
+		if x.abs.Cmp(y.abs) >= 0 {
+			z.abs = z.abs.Sub(x.abs, y.abs)
+		} else {
+			neg = !neg
+			z.abs = z.abs.Sub(y.abs, x.abs)
+		}
+	}
+	z.neg = neg // 0 has no sign
+	return z
+}
+
+// AddUint256 set z to the sum x + y, where y is a uint256, and returns z
+func (z *Int) AddUint256(x *Int, y *uint256.Uint) *Int {
+	if x.neg {
+		if x.abs.Gt(y) {
+			z.abs.Sub(x.abs, y)
+			z.neg = true
+		} else {
+			z.abs.Sub(y, x.abs)
+			z.neg = false
+		}
+	} else {
+		z.abs.Add(x.abs, y)
+		z.neg = false
+	}
+	return z
+}
+
+// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
+func AddDelta(z, x *uint256.Uint, y *Int) {
+	if y.neg {
+		z.Sub(x, y.abs)
+	} else {
+		z.Add(x, y.abs)
+	}
+}
+
+// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
+func AddDeltaOverflow(z, x *uint256.Uint, y *Int) bool {
+	var overflow bool
+	if y.neg {
+		_, overflow = z.SubOverflow(x, y.abs)
+	} else {
+		_, overflow = z.AddOverflow(x, y.abs)
+	}
+	return overflow
+}
+
+// Sub sets z to the difference x-y and returns z.
+func (z *Int) Sub(x, y *Int) *Int {
+	z.initiateAbs()
+
+	neg := x.neg
+	if x.neg != y.neg {
+		// x - (-y) == x + y
+		// (-x) - y == -(x + y)
+		z.abs = z.abs.Add(x.abs, y.abs)
+	} else {
+		// x - y == x - y == -(y - x)
+		// (-x) - (-y) == y - x == -(x - y)
+		if x.abs.Cmp(y.abs) >= 0 {
+			z.abs = z.abs.Sub(x.abs, y.abs)
+		} else {
+			neg = !neg
+			z.abs = z.abs.Sub(y.abs, x.abs)
+		}
+	}
+	z.neg = neg // 0 has no sign
+	return z
+}
+
+// SubUint256 set z to the difference x - y, where y is a uint256, and returns z
+func (z *Int) SubUint256(x *Int, y *uint256.Uint) *Int {
+	if x.neg {
+		z.abs.Add(x.abs, y)
+		z.neg = true
+	} else {
+		if x.abs.Lt(y) {
+			z.abs.Sub(y, x.abs)
+			z.neg = true
+		} else {
+			z.abs.Sub(x.abs, y)
+			z.neg = false
+		}
+	}
+	return z
+}
+
+// Mul sets z to the product x*y and returns z.
+func (z *Int) Mul(x, y *Int) *Int {
+	z.initiateAbs()
+
+	z.abs = z.abs.Mul(x.abs, y.abs)
+	z.neg = x.neg != y.neg // 0 has no sign
+	return z
+}
+
+// MulUint256 sets z to the product x*y, where y is a uint256, and returns z
+func (z *Int) MulUint256(x *Int, y *uint256.Uint) *Int {
```

Comment:
This method was not covered by the testing suite

Reply:
925889d084ce6684d445a933ac0890958838e21a